persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -390,6 +403,31 @@ public void clear() {
       }
    }
 
+   @Override
+   public void clear(IntSet segments) {
+      Connection conn = null;
+      PreparedStatement ps = null;
+      try {
+         String sql = tableManager.getDeleteRowsSqlForSegments(segments.size());
+         conn = connectionFactory.getConnection();
+         ps = conn.prepareStatement(sql);
+         int offset = 0;
+         for (PrimitiveIterator.OfInt segIter = segments.iterator(); segIter.hasNext(); ) {
+            ps.setInt(++offset, segIter.nextInt());
+         }
+         int result = ps.executeUpdate();
+         if (log.isTraceEnabled()) {
+            log.tracef("Successfully removed %d rows.", result);
+         }
+      } catch (SQLException ex) {
+         log.failedClearingJdbcCacheStore(ex);
+         throw new PersistenceException("Failed clearing cache store", ex);
```

Comment:
Should we add the failing segment values to the log/exception message?

Reply:
Sure